components/log-viewer-webui/client/src/api/query.js

```diff
@@ -13,20 +13,21 @@ import axios from "axios";
  */
 
 /**
- * Submits a job to extract the split of an original file that contains a given log event. The file
- * is extracted as a CLP IR file.
+ * Submits a job to extract the stream that contains a given log event. The stream is extracted
+ * either as a CLP IR or a JSON line file.
  *
- * @param {number|string} origFileId The ID of the original file
+ * @param {number} extractJobType Type of the extraction job
+ * @param {number|string} streamId The ID of the original stream
  * @param {number} logEventIdx The index of the log event
  * @param {Function} onUploadProgress Callback to handle upload progress events.
  * @return {Promise<axios.AxiosResponse<ExtractIrResp>>}
```

Comment:
I just noticed this.. 
The ExtractIrResp and ExtractJsonResp are slightly different.

Can i simply define both and indicate that the function can return either?

Reply:
Actually, I just realized that file_split_id is not needed in the stream metadata (it seems that it is not used in the webui). Maybe we can remove it in another PR?

Reply:
Sure. Once we have those defined, I guess we can write:
```
Promise<axios.AxiosResponse<ExtractIrResp | ExtractJsonResp>>
```